addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3094,13 +3094,26 @@ export class OdooEditor extends EventTarget {
             styleSection.querySelector('button span').textContent = activeLabel;
         }
 
+        const isInMedia = this.toolbar.classList.contains('oe-media');
         const linkNode = getInSelection(this.document, 'a');
         const linkButton = this.toolbar.querySelector('#create-link');
         linkButton && linkButton.classList.toggle('active', !!linkNode);
+        // hide unlink button if no link in selection, always hide on media elements
+        const unlinkButton = this.toolbar.querySelector('#unlink');
+        unlinkButton && unlinkButton.classList.toggle('d-none', isInMedia || !linkNode);
```

Comment:
You could do `unlinkButton?.classList.toggle('d-none', isInMedia || !linkNode)`, just like you did for the the `linkButton` and the `linkButtonGoup` in the lines below. (this is optional, as you wish)